apps/hash-frontend/src/components/block-loader/block-loader.tsx

```diff
@@ -141,7 +140,20 @@ export const BlockLoader: FunctionComponent<BlockLoaderProps> = ({
     <RemoteBlock
       blockMetadata={blockMetadata}
       editableRef={editableRef}
-      graphCallbacks={functions}
+      graphCallbacks={{
+        ...functions,
+        updateEntity: async (...args) => {
+          const res = await updateEntity(...args);
+
+          const newBlockSubgraph = await fetchBlockSubgraph(
+            blockEntityTypeId,
+            blockEntityId,
+          );
+          setBlockSubgraph(newBlockSubgraph);
+
+          return res;
+        },
```

Comment:
Why not just add this in the object it was already in? This is making a new object, do we need to?

Reply:
Did that to easily get the types of `...args`, thinking this is a temporary fix. But I can give the type there as well.

Reply:
I think these callback might become stale in the graph hadnler too, if `blockEntityId` changes – can you memoize this based on the value of its dependencies, and in `RemoteBlock` add htis please?

```ts
  useEffect(() => {
    if (graphService) {
      graphService.registerCallbacks(graphCallbacks);
    }
  }, [graphCallbacks, graphService]);
```
this will reregister the callbacks if they change

Reply:
It's an existing defiency with the `useGraphEmbedderService` hook that it does not update the callbacks if they change – it just takes whatever was first passed in. We have a [task](https://app.asana.com/0/1201095311341924/1203257689029614/f) to improve this so that callers don't need to manually re-register callbacks when they change

Reply:
Addressed here https://github.com/hashintel/hash/pull/1955/commits/93feab3f9bee16354259ee6b8f9588005c33c0bf